3단계 - 숫자 클릭

impactrudia opened the pull request:
바쁘신데 빠른 피드백 감사합니다.
이번에도 잘 부탁드립니다 :)

토스트쪽 테스트는 에러가나는데 왜 에러나는지 좀 더 봐야 할 것 같습니다.
androidx.test.espresso.NoMatchingRootException: Matcher '' did not match any of the following roots:
왜 안 되는지 알 수 있을까요? 검색해보고 검색한대로 적용 했는데도 에러가 나서 자문 구합니다.

impactrudia committed:
review : 가독성을 위한 중복된 매직넘버 사용

impactrudia committed:
docs : 테스트 케이스 정리

impactrudia committed:
feat : 입력된_숫자뒤에_숫자를_더_입력하면_연달아서_숫자가_보여야한다

impactrudia committed:
review : 메소드 중괄호 없이 바로 적용

impactrudia committed:
feat : 입력된_값이_없는_상태에서_연산자를_누르면_화면에_아무런_변화가_없어야_한다

impactrudia committed:
refactor : 숫자클릭 리스너 함수 이름, 피연산자 클릭 테스트 코드 분리

impactrudia committed:
refactor : MainActivity 코드 정렬

impactrudia committed:
feat : 숫자가_입력된_상태에서_연산자를_누르면_해당기호가_나타나야한다

impactrudia committed:
docs : 수식에 대한 테스트 케이스 문맥 및 오타

impactrudia committed:
feat : 입력된 값이 없을 때 지우기 버튼을 누르면 화면에 아무런 변화가 없어야한다

impactrudia committed:
feat : 입력된 수식에 끝자리가 숫자이면 숫자 하나만 지워야한다

impactrudia committed:
feat : 입력된 수식에 끝자리가 연산자가 있으면 연산자와 마지막 숫자의 값이 지워져야한다

impactrudia committed:
feat : 연산자가 입력된 상황에서 연산결과를 얻으려고 하면 토스트 메세지가 보여야한다

impactrudia committed:
refactor : 피연산자 코드 줄이기

impactrudia committed:
fix : 연산자 부호가 있을 땐 좌우 공백

impactrudia committed:
fix : 나누기 연산자 / -> ÷

impactrudia committed:
refactor : 필요없는 import, 메소드 이름

impactrudia committed:
fix : 토스트 메세지 id값

galcyurio commented on app/src/main/java/edu/nextstep/camp/calculator/MainActivity.kt:
```diff
@@ -12,41 +14,78 @@ class MainActivity : AppCompatActivity() {
+                    this@MainActivity,
+                    getString(R.string.msg_do_not_match_formula),
+                    Toast.LENGTH_SHORT
+                ).show()
```
별도의 함수로 분리해보세요.

galcyurio commented on app/src/main/java/edu/nextstep/camp/calculator/MainActivity.kt:
```diff
@@ -12,41 +14,78 @@ class MainActivity : AppCompatActivity() {
+        currentView.isEmpty() -> ""
+        isLastTextOperand(currentView) -> currentView.dropLast(1)
+        else -> currentView.dropLast(4)
+    }
```
무언가를 반환하는 함수이니 getXxx()와 같이 이름을 지어주는게 좋을 것 같아요.

galcyurio commented on app/src/main/java/edu/nextstep/camp/calculator/MainActivity.kt:
```diff
@@ -12,41 +14,78 @@ class MainActivity : AppCompatActivity() {
+        currentView.isEmpty() -> ""
+        isLastTextOperand(currentView) -> appendText
+        else -> appendText
+    }
```
무언가를 반환하는 함수이니 getXxx()와 같이 이름을 지어주는게 좋을 것 같아요.

galcyurio reviewed (changes requested):
3단계까지 고생하셨습니다.
코멘트 남겨놓은 몇 가지만 개선해보면 좋겠어요.

impactrudia committed:
review : 별도의 함수로 분리 - showToast

리뷰 내용 : https://github.com/next-step/android-calculator-mvc/pull/15#discussion_r697524625

impactrudia committed:
review : 반환하는 함수에서는 getXXX 메소드 형태 이름

리뷰 내용 : https://github.com/next-step/android-calculator-mvc/pull/15#discussion_r697534109

galcyurio commented on app/src/androidTest/java/edu/nextstep/camp/calculator/MainActivityTest.kt:
```diff
@@ -1,85 +1,110 @@
+            val windowToken = root.decorView.windowToken
+            val appToken = root.decorView.applicationWindowToken
+            if (windowToken === appToken) {
+                //means this window isn't contained by any other windows.
```
이전 리뷰에서 질문을 놓쳤네요.

> 토스트쪽 테스트는 에러가나는데 왜 에러나는지 좀 더 봐야 할 것 같습니다.
androidx.test.espresso.NoMatchingRootException: Matcher '' did not match any of the following roots:
왜 안 되는지 알 수 있을까요? 검색해보고 검색한대로 적용 했는데도 에러가 나서 자문 구합니다.

해당 라인에서 `return true` 해주셔야 합니다.

galcyurio reviewed (approved):
고생하셨습니다. 피드백 잘 반영해주셨네요.
질문해주신 부분은 코멘트로 남겨두었습니다.
